### PR TITLE
feat: Activate ruff rules UP(pyupgrade)

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: src/backend/base/langflow/services/tracing/langwatch.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def __init__(self, trace_name: str, trace_type: str, project_name: str, trace_id
             self.trace = self._client.trace(
                 trace_id=str(self.trace_id),
             )
-            self.spans: dict[str, "ContextSpan"] = {}
+            self.spans: dict[str, ContextSpan] = {}
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/services/tracing/langwatch.py:36]: This might not work because it is not imported if not type checking.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey, @cbornet 

Thanks for this. I was looking forward to getting pyupgrade automated.

File: src/backend/base/langflow/services/tracing/langwatch.py (hunk 1 above)
Comment (cbornet) [reply src/backend/base/langflow/services/tracing/langwatch.py:36]: Oh, right, I forgot that it needs `from __future__ import annotations` to be that way.

**Comment by cbornet** [comment]
I added `from __future__ import annotations` and also set the python min version for ruff, allowing more UP rules to be effective.

**Pull request merged**